### PR TITLE
OKTA-516853 - transformation layer refactor setup 

### PR DESCRIPTION
**Pull request by shuowu-okta**
## Description:

Changes:
* created a new folder (transformernew) with all the refactor code
* added a switch logic in widget/index.ts to turn on new transformation code based on remediation name (IDX_STEP)
* the target for this PR is to have green bacon for Identify step

## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

- Did you verify the change by running downstream monolith artifacts? (YES | NO | UNSURE)

### Issue:

- [OKTA-516853](https://oktainc.atlassian.net/browse/OKTA-516853)

### CODE REVIEW

File: src/v3/src/transformernew/attribute/transform.test.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    const result = { attributes: { autocomplete: 'current-password' } };
+
+    expect(transformField(formfield)).toEqual(result);
+  });
+
+  it('should return uischema object with options.attributes.autocomplete === "current-password" if formfield.name === "newPassword" is present in ion object', () => {
+    const formfield: Input = { name: 'newPassword' };
+
+    const result = { attributes: { autocomplete: 'current-password' } };
```
Comment (lesterchoi-okta) [inline src/v3/src/transformernew/attribute/transform.test.ts:37]: shouldn't autocomplete be `"new-password"`?

File: src/v3/src/transformernew/attribute/transform.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  InputAttributes, 
+  Nullable,
+  FormBag,
+  FieldElement,
+} from '../../types';
+
+const autocompleteValueMap = new Map<string, AutoCompleteValue>([
+  ['identifier', 'username'],
+  ['password', 'current-password'],
+  ['newPassword', 'current-password'],
```
Comment (lesterchoi-okta) [inline src/v3/src/transformernew/attribute/transform.ts:25]:
```suggestion
  ['newPassword', 'new-password'],
```

File: src/v3/src/transformernew/attribute/transform.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  inputName: string,
+): string | undefined => {
+  let isMatch;
+  map.forEach((value: AutoCompleteValue, key: string) => {
+    if (inputName.match(key)?.length) {
+      isMatch = key;
+    }
+  });
+  return isMatch;
+};
```
Comment (lesterchoi-okta) [inline src/v3/src/transformernew/attribute/transform.ts:45]: Reverse lookup? Why is this needed?

File: src/v3/src/transformernew/attribute/transform.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  const { options: { inputMeta: input } } = field;
+  // passcode name is shared with password + totp code types, only differ by secret
+  if (input.name === 'credentials.passcode' && input.secret) {
+    return autocompleteValueMap.get('password') ?? null;
+  }
+  const key = getKeyFromMap(autocompleteValueMap, input.name);
+  if (key) {
+    return autocompleteValueMap.get(key) ?? null;
+  }
+  return null;
```
Comment (lesterchoi-okta) [inline src/v3/src/transformernew/attribute/transform.ts:57]:
```suggestion
  return autocompleteValueMap.get(input.name) ?? null;
```

File: src/v3/src/transformernew/attribute/transform.test.ts (hunk 1 above)
Comment (shuowu-okta) [reply src/v3/src/transformernew/attribute/transform.test.ts:37]: @glenfannin-okta can you confirm?

File: src/v3/src/transformernew/attribute/transform.ts (hunk 3 above)
Comment (shuowu-okta) [reply src/v3/src/transformernew/attribute/transform.ts:46]: @glenfannin-okta ^

File: src/v3/src/transformernew/attribute/transform.test.ts (hunk 1 above)
Comment (glenfannin-okta) [reply src/v3/src/transformernew/attribute/transform.test.ts:37]: Actually, this value does not matter, `newPassword` doesnt seem to be an actual value now that we are using generic remediator version of auth-js. Seems to always be `credentials.passcode` no matter the flow. So this line can be removed completely. We actually update the autocomplete value in the custom transformer because there is no way to decipher what it should be at this level.

File: src/v3/src/transformernew/attribute/transform.ts (hunk 3 above)
Comment (glenfannin-okta) [reply src/v3/src/transformernew/attribute/transform.ts:44]: So, this was done because the keys that are passed to this function are flattened into a string. 

So for example phoneNumber will be presented as `authenticator.phoneNumber` or password will be presented as `credentials.passcode`, otp will either come as `credentials.passcode` or `totp`, etc. 

However, I don't know of every format phoneNumber can come, because it is present when registering as well as when authenticating as well as when setting up a new phone for an authenticator. 

Because of these variables, I just rely on the last part of the field name. So to do that, I have to perform a match on the key of the map (which is what this function does). 

If you are confident the field names are standardized and there are no variances then we can get rid of it but I was not confident in making that assumption which is why I created this lookup function.

File: src/v3/src/transformernew/attribute/transform.ts (hunk 4 above)
Comment (glenfannin-okta) [reply src/v3/src/transformernew/attribute/transform.ts:56]: If this is changed will need to update the keys of the map to include all possible variations of field names (in their flattened format), i.e. `credentials.passcode`, `authenticator.phoneNumber`, `authenticator.firstName`, etc. and be wary of duplicate keys.
